src/Point.ts

```diff
@@ -0,0 +1,46 @@
+const defaultPoint = {
+    x: 0,
+    y: 0
+}
+
+export class Point {
+    public x: number;
+    public y: number;
+
+    constructor();
+    constructor(x: number, y: number);
+    constructor(x?: number, y?: number) {
+        if (!x && !y) {
+            this.x = 0;
+            this.y = 0;
+        }
+
+        if (typeof x === 'number' && typeof y === 'number') {
+            this.x = x;
+            this.y = y;
+        }
+    }
+
+    public distance(): number
+    public distance(other: Point): number
+    public distance(xOrOther?: number | Point, y?: number): number {
```

Comment:
Suppose you could add values by default and don't use const default point at all.

```
public distance(xOrOther: number | Point = 0, y: number = 0): number {
```